### DISCUSSION TITLE
Motivations to initiate this project

### DISCUSSION THREAD

**Discussion by cweisheng322** [iryna-kondr/scikit-llm#71] [General]
May I kindly inquire about the motivations that inspired the authors to initiate this project?

**Comment by iryna-kondr** [comment]
Hi! Our primary objective has been to empower users to easily tackle NLP tasks using LLMs. To achieve this, we decided to use a well-known scikit-learn API. Initially, our support was limited to OpenAI models and a few specific use cases. However, since the release of the library, we've received numerous requests from users for potential integrations. As a result, we have expanded the range of models, including open-source ones, and broadened the variety of use cases. In the near future, we plan to launch version 1.0 of the library and further expand the project's scope.
